akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1093,16 +1093,18 @@ trait FlowOps[+Out, +Mat] {
    *
    * See also [[FlowOps.splitAfter]].
    */
-  def splitWhen(p: Out ⇒ Boolean): SubFlow[Out, Mat, Repr, Closed] = {
+  def splitWhen(p: Out ⇒ Boolean, eagerClose: Boolean = false): SubFlow[Out, Mat, Repr, Closed] = {
```

Comment:
Sure, I just wanted to decide on API and whether those changes are OK or not first.